### PR TITLE
imrovment(code) update code to be compatibility with phpstan level 3

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Update some modules to by compatibility with phpstan level 3

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/authentication/src/Application/Security/Provider/IdUserProvider.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function loadUserByUsername($username)
     /**
      * {@inheritdoc}
      */
-    public function refreshUser(UserInterface $user): User
+    public function refreshUser(UserInterface $user): UserInterface
```
Comment (piotrkreft) [inline module/authentication/src/Application/Security/Provider/IdUserProvider.php:60]: Perhaps the exect instance? We know what it returns 👍🏻

File: module/comment/src/Domain/Event/CommentCreatedEvent.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public function __construct(CommentId $id, UserId $authorId, Uuid $objectId, str
     }
 
     /**
-     * @return CommentId|AggregateId
+     * @return CommentId
      */
     public function getAggregateId(): AggregateId
```
Comment (piotrkreft) [inline module/comment/src/Domain/Event/CommentCreatedEvent.php:58]:
```suggestion
    public function getAggregateId(): CommentId
```

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by wiewiurdp — APPROVED** [review]

File: module/authentication/src/Application/Security/Provider/IdUserProvider.php (hunk 1 above)
Comment (rprzedzik) [reply module/authentication/src/Application/Security/Provider/IdUserProvider.php]: Return value of other method inside this one  is UserIntreface, i change that one to, and now is should be ok.

File: module/multimedia/src/Infrastructure/Persistence/Repository/DbalMultimediaRepository.php [hunk 3]
```diff
@@ -28,19 +26,21 @@ public function __construct(EventStoreManager $manager)
...
+        /** @var AbstractMultimedia|null $aggregate */
         $aggregate = $this->manager->load($id);
         Assert::nullOrIsInstanceOf($aggregate, AbstractMultimedia::class);
 
         return $aggregate;
     }
 
-    public function save(Multimedia $aggregateRoot): void
+    /**
+     * @throws \Doctrine\DBAL\DBALException
```
Comment (wfajczyk) [inline module/multimedia/src/Infrastructure/Persistence/Repository/DbalMultimediaRepository.php:41]: Simplify to fqn

**Review by wfajczyk — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Review by wiewiurdp — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Pull request merged**